File: src/main/java/alfheim/common/spell/earth/SpellMortar.java
package alfheim.common.spell.earth;

import alfheim.api.entity.EnumRace;
import alfheim.api.spell.SpellBase;
import alfheim.common.entity.spell.EntitySpellMortar;
import net.minecraft.entity.EntityLivingBase;

public class SpellMortar extends SpellBase {

	public SpellMortar() {
		super("mortar", EnumRace.GNOME, 7500, 200, 5);
	}

	@Override
	public SpellCastResult performCast(EntityLivingBase caster) {
		SpellCastResult result = checkCast(caster);
		if (result == SpellCastResult.OK) caster.worldObj.spawnEntityInWorld(new EntitySpellMortar(caster.worldObj, caster));
		return result;
	}
}

File: src/main/java/alfheim/common/spell/earth/SpellMortar.kt
package alfheim.common.spell.earth

import alfheim.api.entity.EnumRace
import alfheim.api.spell.SpellBase
import alfheim.common.entity.spell.EntitySpellMortar
import net.minecraft.entity.EntityLivingBase

object SpellMortar: SpellBase("mortar", EnumRace.GNOME, 7500, 200, 5) {
	
	override var damage = 8f
	override var duration = 100 // lifetime
	override var efficiency = 2.0 // speed
	override var radius = 2.0 // AoE
	
	override val usableParams
		get() = arrayOf(damage, duration, efficiency, radius)
	
	override fun performCast(caster: EntityLivingBase): SpellCastResult {
		val result = checkCastOver(caster)
		if (result == SpellCastResult.OK) caster.worldObj.spawnEntityInWorld(EntitySpellMortar(caster.worldObj, caster))
		return result
	}
}